packages/aws-cdk-lib/aws-appsync/test/appsync-eventapi.test.ts

```diff
@@ -0,0 +1,366 @@
+
+// GIVEN
+let stack: cdk.Stack;
+let app: App;
+beforeEach(() => {
+  app = new App({});
+  stack = new cdk.Stack(app);
+});
+
+describe('Appsync Event API Key Authorization', () => {
+  test('Appsync Event API - creates default api key', () => {
+    // WHEN
+    new appsync.EventApi(stack, 'api', { apiName: 'api' });
+
+    // THEN
+    Template.fromStack(stack).resourceCountIs('AWS::AppSync::ApiKey', 1);
+  });
+
+  test('Appsync Event API - allows a single key with no name', () => {
+    // WHEN
+    new appsync.EventApi(stack, 'api', {
+      apiName: 'api',
+      authorizationConfig: {
+        authProviders: [{ authorizationType: appsync.AppSyncAuthorizationType.API_KEY }],
+      },
+    });
+
+    // THEN
+    Template.fromStack(stack).resourceCountIs('AWS::AppSync::ApiKey', 1);
+  });
+
+  test('Appsync Event API - allows multiple keys to be created', () => {
+    // WHEN
+    new appsync.EventApi(stack, 'api', {
+      apiName: 'api',
+      authorizationConfig: {
+        authProviders: [
+          { authorizationType: appsync.AppSyncAuthorizationType.API_KEY, apiKeyConfig: { name: 'first' } },
+          { authorizationType: appsync.AppSyncAuthorizationType.API_KEY, apiKeyConfig: { name: 'second' } },
+        ],
+      },
+    });
+
+    // THEN
+    Template.fromStack(stack).resourceCountIs('AWS::AppSync::ApiKey', 2);
+  });
+
+  test('Appsync Event API - Requires a name when multiple keys are specified', () => {
+    // WHEN and THEN
+    expect(() => {
+      new appsync.EventApi(stack, 'api', {
+        apiName: 'api',
+        authorizationConfig: {
+          authProviders: [
+            { authorizationType: appsync.AppSyncAuthorizationType.API_KEY },
+            { authorizationType: appsync.AppSyncAuthorizationType.API_KEY, apiKeyConfig: { name: 'second' } },
+          ],
+        },
+      });
+    }).toThrow('You must specify key names when configuring more than 1 API key.');
+  });
+});
+
+describe('AppSync Event Api auth configuration', () => {
+  test('Authorization providers are used as allowed modes by default', () => {
+    // WHEN
+    new appsync.EventApi(stack, 'api', {
+      apiName: 'apiWithMultipleProviders',
+      authorizationConfig: {
+        authProviders: [{ authorizationType: appsync.AppSyncAuthorizationType.API_KEY }, { authorizationType: appsync.AppSyncAuthorizationType.IAM }],
+      },
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::AppSync::Api', {
+      Name: 'apiWithMultipleProviders',
+      EventConfig: {
+        ConnectionAuthModes: [{ AuthType: appsync.AppSyncAuthorizationType.API_KEY }, { AuthType: appsync.AppSyncAuthorizationType.IAM }],
+        DefaultPublishAuthModes: [{ AuthType: appsync.AppSyncAuthorizationType.API_KEY }, { AuthType: appsync.AppSyncAuthorizationType.IAM }],
+        DefaultSubscribeAuthModes: [{ AuthType: appsync.AppSyncAuthorizationType.API_KEY }, { AuthType: appsync.AppSyncAuthorizationType.IAM }],
```

Comment:
cc @onlybakam what are your thoughts here?

The CDK is meant to help abstract these extra two steps. If the user only provides the `authProviders` parameter and doesn't explicitly override the `connectionAuthModes`, `defaultPublishAuthModes`, and `defaultSubscribeAuthModes` parameters, we apply all provided `authProviders` as valid connection, publish, and subscribe modes.

This is more for simplicity with the option to override if needed but I can see how this might be confusing. We can provide more clarity in the documentation.

Reply:
Yes, it would be good to update the documentation about the behaviour.